machineconfiguration/v1/doc.go

```diff
@@ -1,5 +1,6 @@
 // +k8s:deepcopy-gen=package,register
 // +groupName=machineconfiguration.openshift.io
 
+// +kubebuilder:validation:Optional
```

Comment:
Why did you chose to add this? Moving fields from being required to optional could be a breaking change so we need to be careful here

Reply:
Historically..."mistakes were made" https://github.com/openshift/api/pull/1453#discussion_r1186430524. Half of our stuff was generated one way, half of it was generated the other. If you'd rather I "invert" it and go through and tag everything optional that was optional instead, I can do that.

Reply:
Thanks, yep, I have now marked all the fields as optional or required. 

I did keep the `+kubebuilder:validation:Optional` for now because:

-  in the existing MCO we a [field that is marked required](https://github.com/openshift/machine-config-operator/blob/40571adccaa3ef49a823abe8d7494bcdc48f2021/install/0000_80_machine-config-operator_01_containerruntimeconfig.crd.yaml#L51) in our CRD but is `omitempty` [in types.go](https://github.com/openshift/machine-config-operator/blob/40571adccaa3ef49a823abe8d7494bcdc48f2021/pkg/apis/machineconfiguration.openshift.io/v1/types.go#L523) (maintaining CRDs by hand...) and that seemed to be the only way to keep parity. 
-  I did generate both ways (with global optional in doc.go and without) and that field was the only difference


(I don't think preserving it matters *that* much, but I figure we can revisit it next time we make an MCO API change and/or do some "opinionated cleanup" on our API?)

Reply:
Any field that isn't explicitly marked as optional or required, but happens to have `omitempty`, becomes optional by the logic of the generators.

Since the CRD had it as required, it should remain as required